### PR TITLE
Add(route): add hacker news user follow

### PR DESCRIPTION
**Pull request by xie-dongping**
## 完整路由地址 / Example for the proposed route(s)

```routes
/hackernews/newcomments/comments_list/
/hackernews/threads/comments_list/dang
/hackernews/submitted/source/dang
```

## 新RSS检查列表 / New RSS Script Checklist
  
- [x] New Route
- [x] Documentation
  - [x] CN
  - [x] EN
- [ ] 全文获取 fulltext
  - [ ] Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [x] 日期和时间 date and time
  - [x] 可以解析 Parsed
  - [x] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added 
- [ ] `Puppeteer`

## 说明 / Note

Add route to subscribe to a certain user in hacker news.

### CODE REVIEW

**Comment by xie-dongping** [comment]
> Successfully generated as following:
> https://rsshub-qo1005cff-diy.vercel.app/hackernews/new_comments/comments_list/ - **Failed**
> https://rsshub-qo1005cff-diy.vercel.app/hackernews/threads/comments_list/dang - Success
> https://rsshub-qo1005cff-diy.vercel.app/hackernews/submitted/source/dang - Success

The route test failed because of a typo, there shouldn't be any underscore in the `newcomments` section.

File: docs/en/programming.md [hunk 1]
```diff
@@ -208,14 +208,20 @@ Section
...
 
 > Default RSS by the website: <https://news.ycombinator.com/rss>, same as `index` section, should be the first choice.
 
 </RouteEn>
 
+### Follow User
+
+Subscribe to the updates (threads and submission) from a paritcular Hacker News user
+
+<RouteEn author="cf020031308 nczitzk xie-dongping" example="/hackernews/threads/comments_list/dang" path="/hackernews/:section?/:type?/:user?" :paramsDesc="['Section, see below, index by default', 'Link, see below, sources by default', 'User, only valid for section threads and submitted']">
```
Comment (TonyRL) [inline docs/en/programming.md:223]:
```suggestion
<RouteEn author="cf020031308 nczitzk xie-dongping" example="/hackernews/threads/comments_list/dang" path="/hackernews/:section?/:type?/:user?" :paramsDesc="['Section, see above, `index` by default', 'Link, see above, `sources` by default', 'User, only valid for section `threads` and `submitted`']" />
```
Please close your Route tag

File: docs/programming.md [hunk 2]
```diff
@@ -291,14 +291,20 @@ GitHub 官方也提供了一些 RSS:
...
 
 > 网站有默认的 RSS：<https://news.ycombinator.com/rss> 内容同 homepage，应优先考虑。
 
 </Route>
 
+### 用户
+
+订阅特定用户的内容
+
+<Route author="cf020031308 nczitzk xie-dongping" example="/hackernews/threads/comments_list/dang" path="/hackernews/:section?/:type?/:user?" :paramsDesc="['内容分区，见下表，默认为 index', '链接类型，见下表，默认为 sources', '设定用户，只在threads和submitted分区有效']" />
```
Comment (TonyRL) [inline docs/programming.md:306]:
```suggestion
<Route author="cf020031308 nczitzk xie-dongping" example="/hackernews/threads/comments_list/dang" path="/hackernews/:section?/:type?/:user?" :paramsDesc="['内容分区，见上表，默认为 `index`', '链接类型，见上表，默认为 `sources`', '设定用户，只在 `threads` 和 `submitted` 分区有效']" />
```

File: docs/programming.md [hunk 3]
```diff
@@ -281,7 +281,7 @@ GitHub 官方也提供了一些 RSS:
 
 ### 分区
 
-<Route author="cf020031308 nczitzk" example="/hackernews" path="/hackernews/:section?/:type?" :paramsDesc="['内容分区，见下表，默认为 index', '链接类型，见下表，默认为 sources']">
+<Route author="cf020031308 nczitzk" example="/hackernews" path="/hackernews/:section?/:type?/:user?" :paramsDesc="['内容分区，见下表，默认为 index', '链接类型，见下表，默认为 sources', '设定用户，只在threads和submitted分区有效']">
```
Comment (TonyRL) [inline docs/programming.md:284]:
```suggestion
<Route author="cf020031308 nczitzk" example="/hackernews" path="/hackernews/:section?/:type?/:user?" :paramsDesc="['内容分区，见下表，默认为 `index`', '链接类型，见下表，默认为 `sources`', '设定用户，只在 `threads` 和 `submitted` 分区有效']">
```
Please add `threads` and `submitted` into the Section table

File: docs/en/programming.md [hunk 4]
```diff
@@ -198,7 +198,7 @@ For instance, the `/github/topics/framework/l=php&o=desc&s=stars` route will gen
 
 ### Section
 
-<RouteEn author="cf020031308 nczitzk" example="/hackernews" path="/hackernews/:section?/:type?" :paramsDesc="['Section, see below, index by default', 'Link, see below, sources by default']">
+<RouteEn author="cf020031308 nczitzk" example="/hackernews" path="/hackernews/:section?/:type?/:user?" :paramsDesc="['Section, see below, index by default', 'Link, see below, sources by default', 'User, only valid for section threads and submitted']">
```
Comment (TonyRL) [inline docs/en/programming.md:201]:
```suggestion
<RouteEn author="cf020031308 nczitzk" example="/hackernews" path="/hackernews/:section?/:type?/:user?" :paramsDesc="['Section, see below, `index` by default', 'Link, see below, `sources` by default', 'User, only valid for section `threads` and `submitted`']">
```

Please add `threads` and `submitted` into the Section table

**Review by TonyRL — CHANGES_REQUESTED** [review]

**Comment by xie-dongping** [comment]
@TonyRL , thanks for the review and the suggested changes. I updated the table according to your suggestion. PTAL

I would like to rebase and squash the PR to one commit, but I'm not sure if it is the normal case so I didn't do so.

File: lib/v2/hackernews/maintainer.js [hunk 5]
```diff
@@ -1,3 +1,3 @@
 module.exports = {
-    '/:section?/:type?': ['nczitzk'],
+    '/:section?/:type?/:user?': ['nczitzk'],
```
Comment (TonyRL) [inline lib/v2/hackernews/maintainer.js:2]:
```suggestion
    '/:section?/:type?/:user?': ['nczitzk', 'xie-dongping'],
```
Please also put your name in it.

**Review by TonyRL — CHANGES_REQUESTED** [review]

File: lib/v2/hackernews/maintainer.js (hunk 5 above)
Comment (xie-dongping) [reply lib/v2/hackernews/maintainer.js]: Done

**Comment by xie-dongping** [comment]
Added myself as maintainer and also rebased/squashed to one commit.

**Pull request merged**